src/svlangserver.ts

```diff
@@ -95,9 +95,9 @@ let svhiercalculator: SystemVerilogHierarchyCalculator = new SystemVerilogHierar
 let settings = default_settings;
 
 connection.onInitialize((params: InitializeParams) => {
-    hasConfigurationCapability = !!(params.capabilities.workspace && !!params.capabilities.workspace.configuration);
-    hasShowWindowCapability = !!(params.capabilities.window && params.capabilities.window.showDocument);
-    clientName = !!params.clientInfo ? params.clientInfo.name : undefined;
+    hasConfigurationCapability = params.capabilities.workspace?.configuration == true;
+    hasShowWindowCapability = params.capabilities.window?.showDocument?.support == true;
+    clientName = params.clientInfo?.name;
```

Comment:
Some other stylistic changes, using the [optional chaining operator `?.`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), happy to revert the unrelated changes though